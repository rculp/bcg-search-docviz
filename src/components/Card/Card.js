import React from 'react';
import uuid from 'uuid/v1';
import { Card } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import './Card.scss';

Card.PracticeArea = ({ practiceAreas }) => (
  <span className="practiceAreas">
    {
      practiceAreas.map((pa, index) => (
        <span key={uuid()}>{ index > 0 && <span>&nbsp;&bull;&nbsp;</span> }{pa}</span>
      ))
    }
  </span>
);

Card.MatchPercentage = ({ relevancyScore }) => (
  <span className="matchPercentage">{ relevancyScore * 100 }% MATCH</span>
);

Card.PracticeArea.propTypes = {
  practiceAreas: PropTypes.arrayOf(PropTypes.string)
};

Card.PracticeArea.defaultProps = {
  practiceAreas: []
};

Card.MatchPercentage.propTypes = {
  relevancyScore: PropTypes.number
};

Card.MatchPercentage.defaultProps = {
  relevancyScore: 0
};

export default Card;
