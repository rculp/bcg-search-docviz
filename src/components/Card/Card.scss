.ui.card {
  &.green,
  &.green:hover {
    border-top: 4px solid #3fad93;
    box-shadow: 0 1px 3px 0 #d4d4d5, 0 0 0 1px #d4d4d5;
  }

  .practiceAreas {
    display: inline-block;
    color: #a9a9a9;
  }

  .matchPercentage {
    display: inline-block;
    color: #3fad93;
    float: right;
  }
}
