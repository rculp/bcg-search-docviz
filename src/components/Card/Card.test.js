import { shallow } from 'enzyme';

import React from 'react';

import Card from './Card';

describe('Card', () => {
  it('renders and matches our snapshot', () => {
    const component = shallow(
      <Card />
    );
    expect(component).toMatchSnapshot();
  });

  it('renders and matches our snapshot', () => {
    const component = shallow(
      <Card.PracticeArea>Area</Card.PracticeArea>
    );
    expect(component).toMatchSnapshot();
  });

  it('renders and matches our snapshot', () => {
    const component = shallow(
      <Card.MatchPercentage>100%</Card.MatchPercentage>
    );
    expect(component).toMatchSnapshot();
  });
});