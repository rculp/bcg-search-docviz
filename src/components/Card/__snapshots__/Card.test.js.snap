// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders and matches our snapshot 1`] = `
<Card>
  Card
</Card>
`;
